src/modules/blockchainApplication/manage/store/reducer.js

```diff
@@ -36,20 +36,32 @@ export const pins = (state = initialState.pins, { type, chainId }) => {
 /**
  *
  * @param {Object} state
- * @param {type: String, data: Object} action
+ * @param {type: String, app: Object, apps: Object, chainId: String} action
  */
-export const applications = (state = initialState.applications, { type, application, chainId }) => {
+export const applications = (state = initialState.applications, { type, app, apps, chainId }) => {
   switch (type) {
     case actionTypes.addApplicationByChainId:
       // In cases where a new node for an existing application is being added,
       // the new node url should be appended to the apis array of the application
-      if (application.chainID in state) {
-        state[application.chainID].serviceURLs.push(application.serviceURLs);
+      if (app.chainID in state) {
+        state[app.chainID].serviceURLs.push(app.serviceURLs);
       } else {
-        state[application.chainID] = application;
+        state[app.chainID] = app;
       }
       return state;
 
+    case actionTypes.setApplications: {
+      apps.forEach((application) => {
```

Comment:
Couldnt this be written in an immutable way with just a `.map(`?

Reply:
I guess so but I think just iterating through the array list and re-consuming the checks that are done when setting a single application was way easier.

Reply:
@oskarleonard good catch 

@eniolam1000752 redux is imotable we shoulnt manipulate the state in this way we need to return a new object like below

```js
 if (app.chainID in state) {
cons newState = {
 ...state
}
newState[app.chainID].serviceURLs.push(app.serviceURLs)
       return newState
      } else {
       return {
        ...state,
        [app.chainID]: app;
        }
      }
```

Reply:
more info here
https://stackoverflow.com/questions/43759080/what-could-happen-if-modifying-state-directly-inside-a-redux-reducer